File: vb/HoneydewTestProjectVB/Generics.vb
Imports HoneydewTestProjectVB.A

Friend Class Generics
    Public Sub Method(ByVal c1Params As List(Of C1))
        Dim x = c1Params.Single().X
    End Sub

    Public Sub Method(ByVal d1Params As List(Of D1))
        Dim x = d1Params
    End Sub
End Class

File: csharp/HoneydewTestProject/Generics.cs
using System.Collections.Generic;
using System.Linq;
using HoneydewTestProject.A;

namespace HoneydewTestProject
{
    internal class Generics
    {
        public void Method(List<C1> c1Params)
        {
            var x = c1Params.Single().X;
        }

        public void Method(List<D1> d1Params)
        {
            var x = d1Params;
        }
    }
}